softgroup/basis/common-token/src/main/java/com/softgroup/token/impl/TokenService.java

```diff
@@ -0,0 +1,137 @@
+package com.softgroup.token.impl;
+
+import com.softgroup.token.api.JwtUserIdentifier;
+import com.softgroup.token.api.TokenGeneratorService;
+import com.softgroup.token.api.TokenType;
+import com.softgroup.token.api.UserIdentifier;
+import io.jsonwebtoken.*;
+import org.springframework.stereotype.Service;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import static com.softgroup.token.api.TokenType.*;
+
+/**
+ * Created by user on 17.03.2017.
+ */
+
+
+@Service
+public class TokenService implements TokenGeneratorService {
+
+    private KeyFactory keyFactory = new KeyFactory();
+
+    //Long term token lifetime in month
+    private int longTermTokenLifetime = 12;
+    //Short term token lifetime in minute
+    private int shortTermTokenLifetime = 10;
+
+    public String createLongTermToken(String deviceId, String userId) {
+        if (deviceId == null || userId == null)
+            return null;
```

Comment:
@Kun-Kun Can we create token with empty data? If not - than null token is wrong result. If yes - your method must return token.